### ISSUE TITLE
Use devel client tools when applicable

### ISSUE THREAD

**Issue by mbologna** [uyuni-project/sumaform#717]
Introduce a configuration parameter to allow the usage of devel client tools on top of the client tools (devel tools must have higher priority than regular client tools) in automated/testing deployments.

Via Salt in sumaform:

- [ ] Add custom channels with devel client tools
- [ ] Sync custom channels
- [ ] Associate the custom channels to onboarded minions

**Comment by juliogonzalez** [comment]
> devel tools must have higher priority than regular client tools

Good point!

This could be a problem ATM. Do we allow vendor change at Uyuni/SUSE Manager? Because SUSE is the vendor for Releaded and Beta client tools, while IBS is the vendor for the Devel client tools.

**Comment by mbologna** [comment]
> This could be a problem ATM. Do we allow vendor change at Uyuni/SUSE Manager? Because SUSE is the vendor for Releaded and Beta client tools, while IBS Devel:Galaxy is the vendor for the Devel client tools.

Yes, we allow equivalent vendors that include SUSE (only for SUSE Manager), OpenSUSE, and Devel:Galaxy :

https://github.com/uyuni-project/sumaform/blob/master/salt/repos/default.sls#L325-L340

**Comment by juliogonzalez** [comment]
Good to know.

Then we don't really need to worry about  `<version.release>`, at least for SLE and RES/RHEL/CentOS/Oracle.

The Head projects for such clients have:
```
Release: 401.<CI_CNT>.<B_CNT>.develHead
```
And at 4.0:
```
Release: 400.<CI_CNT>.<B_CNT>.devel40
```
So release should always be greater, even when version is the same.

Ubuntu could be another story, however. We don't have this, and I don't know if it will work. I will try tomorrow.
